src/polymer-ready.js

```diff
@@ -17,38 +17,51 @@ function showPageAction() {
 };
 
 chrome.runtime.onConnect.addListener(function(port) {
+  var allCustomElements = [];
+
+  function isCustomElement(el) {
+    const isAttr = el.getAttribute('is');
+    return el.localName.includes('-') || isAttr && isAttr.includes('-');
+  }
+
+  function findAllCustomElements(nodes) {
+    nodes.forEach((node) => {
+      if (isCustomElement(node)) { allCustomElements.push(node); }
+      if (node.shadowRoot) { findAllCustomElements(node.shadowRoot.querySelectorAll('*')); }
+    });
+  }
+
   port.onMessage.addListener(function(msg) {
     switch(msg.action) {
 
       case 'get-custom-elements':
         // Get all custom elements.
-        customElements = Array.prototype.slice.call(document.querySelectorAll('* /deep/ *')).filter(function(element) {
-          return element.localName.indexOf('-') != -1 || element.getAttribute('is');
-        });
-        if (customElements.length === 0) {
-          return;
-        }
+        findAllCustomElements(document.querySelectorAll('*'));
```

Comment:
@hdelossantos @beaufortfrancois 

Is this safer to ensure the arguments will be always in the form of Array?

```patch
- findAllCustomElements(document.querySelectorAll('*'));
+ findAllCustomElements([...document.querySelector('*')]);
```